homeassistant/components/wink.py

```diff
@@ -23,19 +24,60 @@
 DOMAIN = 'wink'
 
 SUBSCRIPTION_HANDLER = None
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+CONF_USER_AGENT = 'user_agent'
+CONF_OATH = 'oath'
+CONF_DEFINED_BOTH_MSG = 'Remove access token to use oath2.'
+CONF_MISSING_OATH_MSG = 'Missing oath2 credentials.'
 
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
-        vol.Required(CONF_ACCESS_TOKEN): cv.string,
-    }),
+        vol.Inclusive(CONF_EMAIL, CONF_OATH,
+                      msg=CONF_MISSING_OATH_MSG): cv.string,
+        vol.Inclusive(CONF_PASSWORD, CONF_OATH,
+                      msg=CONF_MISSING_OATH_MSG): cv.string,
+        vol.Inclusive(CONF_CLIENT_ID, CONF_OATH,
+                      msg=CONF_MISSING_OATH_MSG): cv.string,
+        vol.Inclusive(CONF_CLIENT_SECRET, CONF_OATH,
+                      msg=CONF_MISSING_OATH_MSG): cv.string,
+        vol.Exclusive(CONF_EMAIL, CONF_OATH,
+                      msg=CONF_DEFINED_BOTH_MSG): cv.string,
+        vol.Exclusive(CONF_ACCESS_TOKEN, CONF_OATH,
+                      msg=CONF_DEFINED_BOTH_MSG): cv.string,
+        vol.Optional(CONF_USER_AGENT): cv.string
+    })
 }, extra=vol.ALLOW_EXTRA)
 
 
 def setup(hass, config):
     """Setup the Wink component."""
     import pywink
+
+    try:
+        user_agent = config[DOMAIN][CONF_USER_AGENT]
+    except KeyError:
+        user_agent = None
+
+    if user_agent:
+        pywink.set_user_agent(user_agent)
+
     from pubnub import Pubnub
-    pywink.set_bearer_token(config[DOMAIN][CONF_ACCESS_TOKEN])
+    try:
+        access_token = config[DOMAIN][CONF_ACCESS_TOKEN]
```

Comment:
You can replace this `try…catch` block with:

```python
access_token = config[DOMAIN].get(CONF_ACCESS_TOKEN)
```

Docs: `get(key, default=None)` returns `default` if no key found.